elastic/kibana
# On search source error, show 'view details' action that opens request in inspector

Closes https://github.com/elastic/kibana/issues/167904

PR updates `EsError` with `getActions` method that returns "View details" button. Clicking "View details" opens inspector to request that failed. PR updates Discover and maps to display EsError action.

PR does not update lens to display "View details". Chatted with @drewdaemon and the implementation path is more involved. This will be completed in another PR.

### Test setup
1. install sample web logs

### Test discover with EsError
1. open discover 
2. Add filter
    ```
    {
      "error_query": {
        "indices": [
          {
            "error_type": "exception",
            "message": "local shard failure message 123",
            "name": "kibana_sample_data_logs"
          }
        ]
      }
    }
    ```
3. Verify `View details` action is displayed and clicking action opens inspector
    <img width="300" alt="Screenshot 2023-11-07 at 12 53 31 PM" src="https://github.com/elastic/kibana/assets/373691/6b43e9c8-daab-4782-876e-ded6958d15cf">

### Test search embeddable with EsError
1. create new dashboard. Add saved search from `kibana_sample_data_logs` data view 
2. Add filter
    ```
    {
      "error_query": {
        "indices": [
          {
            "error_type": "exception",
            "message": "local shard failure message 123",
            "name": "kibana_sample_data_logs"
          }
        ]
      }
    }
    ```
3. Verify `View details` action is displayed and clicking action opens inspector
    <img width="300" alt="Screenshot 2023-11-07 at 12 55 46 PM" src="https://github.com/elastic/kibana/assets/373691/5ebe37c6-467a-4d72-89e3-21fc53f59d89">

### Test discover with PainlessError
<img width="300" alt="Screenshot 2023-11-07 at 12 52 51 PM" src="https://github.com/elastic/kibana/assets/373691/6d17498f-657c-46e8-86e8-dde461599267">

### Test Maps error
1. create new map
2. Add `documents` layer
3. Set scaling to "limit to 10000"
4. Add filter
    ```
    {
      "error_query": {
        "indices": [
          {
            "error_type": "exception",
            "message": "local shard failure message 123",
            "name": "kibana_sample_data_logs"
          }
        ]
      }
    }
    ```
5. Verify "View details" button is displayed in maps legend in error callout
    <img width="500" alt="Screenshot 2023-11-08 at 12 07 42 PM" src="https://github.com/elastic/kibana/assets/373691/2eb2cc41-0919-49a3-9792-fda9707973cb">
